Lib/ldap/dn.py

```diff
@@ -120,3 +120,33 @@ def is_dn(s,flags=0):
     return False
   else:
     return True
+
+
+def compare_dn(a, b):
+    """
+    Compares two distinguished names and return True if they are equal.
+    """
+    case_insensitive_attributes = [attr.lower() for attr in compare_dn.case_insensitive_attributes]
+
+    def normalize(attr, value):
+        if attr.lower() in case_insensitive_attributes:
+            value = value.lower()
```

Comment:
This assumes trivial lower case conversation. LDAP servers can have more complex and language specific conversation and comparsion.